### ISSUE TITLE
salt-cloud create without profile broken

### ISSUE THREAD

**Issue by codehotter** [saltstack/salt#26369]
Recently it's no longer possible to do a create without a profile.

```
import salt.cloud
import json

client = salt.cloud.CloudClient(path='/etc/salt/cloud')

create_args = {
    "image": "ami-e4ff5c93", # Centos 7 HVM
    "size": "t2.micro",
}
ret = client.create("ec2-ireland-test", "created_without_profile", **create_args)
print(json.dumps(ret, indent=4))
```

Returns

```
{
    "created_without_profile": {}
}
```

And no machine is created. After the below patch it works again. (not sure if correct)

```
diff --git a/salt/cloud/clouds/ec2.py b/salt/cloud/clouds/ec2.py
index dab6940..7f67245 100644
--- a/salt/cloud/clouds/ec2.py
+++ b/salt/cloud/clouds/ec2.py
@@ -2165,7 +2165,7 @@ def create(vm_=None, call=None):
 
     try:
         # Check for required profile parameters before sending any API calls.
-        if config.is_profile_configured(__opts__,
+        if vm_['profile'] and config.is_profile_configured(__opts__,
                                         __active_provider_name__ or 'ec2',
                                         vm_['profile']) is False:
             return False
```

salt-cloud --versions-report

```
Salt Version:
            Salt: 2015.8.0rc3-1202-gc34c6b9
 
Dependency Versions:
 Apache Libcloud: 0.17.0
          Jinja2: 2.7.2
        M2Crypto: 0.21.1
            Mako: Not Installed
          PyYAML: 3.10
           PyZMQ: 14.3.1
          Python: 2.7.5 (default, Jun 24 2015, 00:41:19)
            RAET: Not Installed
         Tornado: 4.2
             ZMQ: 3.2.5
           ioflo: Not Installed
         libnacl: Not Installed
    msgpack-pure: Not Installed
  msgpack-python: 0.4.6
        pycrypto: 2.6.1
         timelib: Not Installed
 
System Versions:
            dist: centos 7.1.1503 Core
         machine: x86_64
         release: 3.10.0-123.8.1.el7.x86_64
          system: CentOS Linux 7.1.1503 Core
```

**Comment by rallytime** [comment]
Good catch @codehotter! That fix looks great - many thanks for your detective work. Your patch will fix the bug in the ec2 driver, but I have caused this bug in a bunch of other drivers as well. I'll clean them all up today. 

Apologies for the troubles. I caused this regression trying to fix another bug where we're trying to be more helpful about missing provider/profile configs.

**Comment by cachedout** [comment]
Closed via #26383
